packages/elements/src/components/ino-datepicker/validation.ts

```diff
@@ -3,58 +3,191 @@ import flatpickr from 'flatpickr';
 
 const NUMBERS_WITH_SPECIAL_CHARS = /(\d[^a-z]+)/g;
 
-export const validateRange = (value: string, format: string): boolean =>
-  !value
-    .match(NUMBERS_WITH_SPECIAL_CHARS)
-    .map((match) => validateFormat(match.trim(), format))
-    .includes(false);
-
-export const validateSingle = (
-  value: string,
-  format: string,
-  minDateStr?: string,
-  maxDateStr?: string
-): boolean =>
-  validateMin(value, format, minDateStr) &&
-  validateMax(value, format, maxDateStr) &&
-  validateFormat(value, format);
-
-export const validateMin = (
-  value: string,
-  format: string,
-  minDateStr?: string
-): boolean => {
-  if (!minDateStr) {
-    return true;
-  }
-  const parsedDate: Date = parseDate(value, format);
-  const minDate = flatpickr.parseDate(minDateStr, format);
-  return isEqual(parsedDate, minDate) || isAfter(parsedDate, minDate);
+type Props = {
+  disabled: boolean;
+  required: boolean;
+  dateFormat: string;
+  isRanged: boolean;
+  minDate?: string;
+  maxDate?: string;
 };
 
-export const validateMax = (
-  value: string,
-  format: string,
-  maxDateStr?: string
-): boolean => {
-  if (!maxDateStr) {
+export class DateValidator {
+  private _isDisabled: boolean;
+  private _isRequired: boolean;
+  private _dateFormat: string;
+  private _minDate?: Date;
+  private _maxDate?: Date;
+  private _isRanged: boolean;
+
+  constructor({
+    disabled,
+    required,
+    dateFormat,
+    isRanged,
+    minDate,
+    maxDate,
+  }: Props) {
+    this._isDisabled = disabled;
+    this._isRequired = required;
+    this._dateFormat = dateFormat;
+    this._isRanged = isRanged;
+    this.minDate = minDate;
+    this.maxDate = maxDate;
+  }
+
+  /**
+   * Validates the date in terms of format, require-state, disabled-state minimum date and maximum date.
+   * @param value The value to be validated
+   * @return true if the date is valid, false if the date is invalid
+   */
+  public validate = (value?: string): boolean => {
+    if (this._isDisabled) {
+      return true;
+    }
+
+    if (!value && this._isRequired) {
+      return false;
+    }
+
+    if (!this.validateFormat(value)) {
```

Comment:
validateFormat and validateMax/validateMin has to be performed in parallel. Otherwise values outter scope  would produce valid date input

Reply:
e.g.

if(!value) return this._isRequired;

if(value)   return  validateMin&& validateMax && validateFormat;

see former implementation

Reply:
Do you have a concrete example where the current implementation would fail?
I could write it the way you described but I think this way the readability is better.

Reply:
Sorry my bad. I realized that the logic work as expected. The Issue that occured is due to parsing date in min and max setter. If the `minDate` is initialized asynchronous in any framework the value may be initially 'undefined' and the datepicker errors out. Maybe we should adjust the setter logic to only parse Values if value is not undefined